src/faction_camp.cpp

```diff
@@ -190,458 +192,477 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
-            mission_key.push( title_e, dir + _( " Expansion Upgrade" ), dir );
-            std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_upgrade_exp_" + dir );
+            entry = om_upgrade_description( bldg_exp );
+            mission_key.add_start( title_e, dir + _( " Expansion Upgrade" ), dir, entry,
+                                   npc_list.empty() );
             if( !npc_list.empty() ) {
                 entry = _( "Working to upgrade your expansions!\n" );
                 bool avail = false;
                 for( auto &elem : npc_list ) {
                     avail |= update_time_left( entry, elem );
                 }
                 entry = entry + _( "\n \nDo you wish to bring your allies back into your party?" );
-                mission_key.text["Recover Ally, " + dir + " Expansion"] = entry;
-                mission_key.push( "Recover Ally, " + dir + " Expansion",
-                                  _( "Recover Ally, " ) + dir + _( " Expansion" ), dir, true, avail );
+                mission_key.add_return( "Recover Ally, " + dir + " Expansion",
+                                        _( "Recover Ally, " ) + dir + _( " Expansion" ), dir,
+                                        entry, avail );
             }
         }
 
         if( bcp->has_level( "garage", 1, dir ) ) {
-            std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_exp_chop_shop_" + dir );
+            comp_list npc_list = companion_list( p, "_faction_exp_chop_shop_" + dir );
             std::string title_e = dir + " Chop Shop";
-            mission_key.text[title_e] = _( "Notes:\n"
-                                           "Have a companion attempt to completely dissemble a vehicle into components.\n \n"
-                                           "Skill used: mechanics\n"
-                                           "Difficulty: 2 \n"
-                                           "Effects:\n"
-                                           "> Removed parts placed on the furniture in the garage.\n"
-                                           "> Skill plays a huge role to determine what is salvaged.\n \n"
-                                           "Risk: None\n"
-                                           "Time: Skill Based \n" );
-            mission_key.push( title_e, dir + _( " Chop Shop" ), dir, false, npc_list.empty() );
+            entry = _( "Notes:\n"
+                       "Have a companion attempt to completely dissemble a vehicle into "
+                       "components.\n \n"
+                       "Skill used: mechanics\n"
+                       "Difficulty: 2 \n"
+                       "Effects:\n"
+                       "> Removed parts placed on the furniture in the garage.\n"
+                       "> Skill plays a huge role to determine what is salvaged.\n \n"
+                       "Risk: None\n"
+                       "Time: Skill Based \n" );
```

Comment:
Time isn't skill based anymore is it?

Reply:
Maybe just comment out the line entirely until it's true?

Reply:
Time is 5 days fixed for now and the message says that.

Reply:
Oops, I fixed this in a separate PR.